landlab/components/river_bed_dynamics/river_bed_dynamics.py

```diff
@@ -0,0 +1,3433 @@
+
+        # Calculating D50 at each node - Grain sizes in Psi scale
+        grain_size_Psi_scale_D = np.flip(np.log2(grain_size_D), axis=0)
+        # Frequency of each grain size in each node
+        grain_size_D_equivalent_frequency_cumulative = np.hstack(
+            (
+                np.zeros(
+                    [
+                        self._bed_surface__grain_size_distribution_location_node.shape[
+                            0
+                        ],
+                        1,
+                    ]
+                ),
+                np.cumsum(np.flip(grain_size_D_equivalent_frequency, axis=1), axis=1),
+            )
+        )
+
+        # Finds the index of the grain size smaller than 50% in each node
+        i0 = np.argmin(grain_size_D_equivalent_frequency_cumulative <= 0.5, axis=1) - 1
+        # Finds the index of the grain size larger than 50% in each node
+        i1 = np.argmax(grain_size_D_equivalent_frequency_cumulative > 0.5, axis=1)
+        nodes_list = np.arange(0, self._grid.number_of_nodes)
+
+        grain_size_Psi_scale_D50 = grain_size_Psi_scale_D[i0] + (
+            (grain_size_Psi_scale_D[i1] - grain_size_Psi_scale_D[i0])
+            / (
+                grain_size_D_equivalent_frequency_cumulative[nodes_list, i1]
+                - grain_size_D_equivalent_frequency_cumulative[nodes_list, i0]
+            )
+        ) * (0.5 - grain_size_D_equivalent_frequency_cumulative[nodes_list, i0])
+        D50 = 2**grain_size_Psi_scale_D50  # Median grain size in each node
+
+        # Calculating the geometric mean and standard deviation
+        # Equivalent grain sizes in Psi scale
+        Psi_grain_size_D_equivalent = np.log2(grain_size_D_equivalent)
+        Psi_grain_size_D_equivalent_mean = np.sum(
+            grain_size_D_equivalent_frequency * Psi_grain_size_D_equivalent, axis=1
+        )
+        grain_size_geometric_mean = 2**Psi_grain_size_D_equivalent_mean
+        # Equivalent grain sizes in Psi scale at each node
+        Psi_grain_size_D_equivalent = np.tile(
+            Psi_grain_size_D_equivalent, (self._grid.number_of_nodes, 1)
+        )
+        Psi_grain_size_D_equivalent_mean = np.reshape(
+            Psi_grain_size_D_equivalent_mean, [self._grid.number_of_nodes, 1]
+        )
+        grain_size_geometric_standard_deviation = 2 ** np.sqrt(
+            np.sum(
+                ((Psi_grain_size_D_equivalent - Psi_grain_size_D_equivalent_mean) ** 2)
+                * grain_size_D_equivalent_frequency,
+                axis=1,
+            )
+        )
+
+        # We save all data into the grid so it can be shared with different
+        # functions and components
+        # Bed grain sizes frequency in each node
+        self._bed_surface__grain_size_distribution_node = (
+            grain_size_D_equivalent_frequency
+        )
+        self._bed_surface__grain_size_distribution_original_node = copy.deepcopy(
+            grain_size_D_equivalent_frequency
+        )
```

Comment:
I don't think you need to use ``deepcopy`` here.``grain_size_D_equivalent_frequency.copy()`` should work just fine.

Reply:
Thanks, done!